### PR TITLE
Change symbol type from hash to string in contract

### PR DESCRIPTION
**Pull request by satawatnack**

### CODE REVIEW

File: contracts/std_ref/lib.rs [hunk 1]
```diff
@@ -19,8 +20,8 @@ mod std_ref {
         admin: AccountId,
         /// Mapping of the granted relayers
         relayers: Mapping<AccountId, ()>,
-        /// Mapping from hash of symbol to price datum
-        ref_data: Mapping<Hash, RefDatum>,
+        /// Mapping from String of symbol to price datum
```
Comment (taobun) [inline contracts/std_ref/lib.rs:23]: Fix comment

File: contracts/std_ref/lib.rs [hunk 2]
```diff
@@ -61,7 +62,7 @@ mod std_ref {
             }
 
             set_code_hash(&code_hash)
-                .unwrap_or_else(|err| panic!("Failed to set code hash due to {:?}", err));
+                .unwrap_or_else(|err| panic!("Failed to set code String due to {:?}", err));
```
Comment (taobun) [inline contracts/std_ref/lib.rs:65]: fix to hash please

**Review by taobun — CHANGES_REQUESTED** [review]

File: contracts/std_ref/lib.rs [hunk 3]
```diff
@@ -15,30 +16,30 @@ mod std_ref {
...
     pub struct StandardReference {
-        /// Address of admin who can grant/revoke relayers
+        // Address of admin who can grant/revoke relayers
         admin: AccountId,
-        /// Mapping of the granted relayers
+        // Mapping of the granted relayers
         relayers: Mapping<AccountId, ()>,
-        /// Mapping from hash of symbol to price datum
-        ref_data: Mapping<Hash, RefDatum>,
+        // Mapping from String of symbol to price datum
```
Comment (RogerKSI) [inline contracts/std_ref/lib.rs:23]:
```suggestion
        // Mapping from string of symbol to price datum
```

**Review by RogerKSI — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

**Review by RogerKSI — APPROVED** [review]

File: contracts/std_ref/lib.rs [hunk 4]
```diff
@@ -19,8 +20,8 @@ mod std_ref {
         admin: AccountId,
         /// Mapping of the granted relayers
         relayers: Mapping<AccountId, ()>,
-        /// Mapping from hash of symbol to price datum
-        ref_data: Mapping<Hash, RefDatum>,
+        /// Mapping from string of symbol to price datum
```
Comment (taobun) [inline contracts/std_ref/lib.rs:23]: Mapping from symbol to ...

File: contracts/std_ref/lib.rs [hunk 5]
```diff
@@ -140,7 +144,7 @@ mod std_ref {
         #[ink(message)]
         pub fn get_reference_data_bulk(
             &mut self,
-            symbol_pair: Vec<(Hash, Hash)>,
+            symbol_pair: Vec<(String, String)>,
```
Comment (taobun) [inline contracts/std_ref/lib.rs:147]: Should we change to symbol_pairs?

File: contracts/std_ref/lib.rs [hunk 6]
```diff
@@ -129,9 +130,12 @@ mod std_ref {
 
         /// Returns the reference data for a given symbol
         #[ink(message)]
-        pub fn get_reference_data(&mut self, symbol_pair: (Hash, Hash)) -> Result<ReferenceData> {
-            let base = self.get_ref_data(&symbol_pair.0)?;
-            let quote = self.get_ref_data(&symbol_pair.1)?;
+        pub fn get_reference_data(
+            &mut self,
+            symbol_pairs: (String, String),
```
Comment (taobun) [inline contracts/std_ref/lib.rs:135]: Use symbol_pair for 1 pair

**Review by taobun — APPROVED** [review]

**Pull request merged**